### ISSUE TITLE
can-i-deploy command returns true even though version was not verified against given tag

### ISSUE THREAD

**Issue by tinexw** [pact-foundation/pact_broker-client#33]
# Pre issue-raising checklist

I have already (please mark the applicable with an `x`):

* [x] Upgraded to the latest Pact Broker OR
* [ ] Checked the CHANGELOG to see if the issue I am about to raise has been fixed

## Software versions

* **pact-broker version:** 2.23.3
* **pact broker client details:** pact-ruby-standalone CLI v 1.47.1 and pact-jvm v 3.5.19

## Expected behaviour

I have one consumer `de.tine.foo` and two providers `de.tine.bar` and `de.tine.baz`.

They have the following matrix:
<img width="747" alt="matrix-bar" src="https://user-images.githubusercontent.com/7386945/42133850-e6c09076-7d30-11e8-9bbf-6c815a33308f.png">
<img width="739" alt="matrix-baz" src="https://user-images.githubusercontent.com/7386945/42133851-e9a38046-7d30-11e8-8fd3-21170dd8b724.png">
 
If I run 

```
can-i-deploy -a de.tine.foo -e 3.0.0 --to prod
```
I expect that the command fails because version `3.0.0` of `de.tine.foo` was not verified against the latest `prod` tag of `de.tine.bar`.

## Actual behaviour

The command succeeds. Output:

````
Computer says yes \o/ 

CONSUMER    | C.VERSION | PROVIDER    | P.VERSION | SUCCESS?
------------|-----------|-------------|-----------|---------
de.tine.foo | 3.0.0     | de.tine.baz | 20.0.0    | true    

All verification results are published and successful

````

## Steps to reproduce
Code to reproduce in https://github.com/tinexw/pact-broker-example. Please see README in repo for details.
Data was already published once to the <https://test.pact.dius.com.au> broker, s. <https://test.pact.dius.com.au/matrix/provider/de.tine.bar/consumer/de.tine.foo> and <https://test.pact.dius.com.au/matrix/provider/de.tine.baz/consumer/de.tine.foo>

The whole issue seems to be caused by the fact that an old provider version is used again. This should actually not happen in a correct setup. However, it did happen in my organization because of a misconfiguration of the provider. But I think that the `can-i-deploy` command should prevent us from deploying even in those cases. Additionally, it might be a valid configuration after all in case the provider does a rollback to a previous version.

## Relevent log files
Output when running with ``--verbose`` option. 

```
opening connection to test.pact.dius.com.au:443...
opened
starting SSL for test.pact.dius.com.au:443...
SSL established
<- "GET /matrix?q[][pacticipant]=de.tine.foo&q[][version]=3.0.0&latestby=cvp&latest=true&tag=prod HTTP/1.1\r\nAccept: application/hal+json, application/json\r\nAuthorization: Basic <snip>==\r\nConnection: close\r\nHost: test.pact.dius.com.au\r\n\r\n"
-> "HTTP/1.1 200 OK\r\n"
-> "Content-Type: application/hal+json;charset=utf-8\r\n"
-> "Date: Sun, 01 Jul 2018 11:24:38 GMT\r\n"
-> "Server: Webmachine-Ruby/1.5.0 Rack/1.3\r\n"
-> "Status: 200 OK\r\n"
-> "Vary: Accept\r\n"
-> "X-Content-Type-Options: nosniff\r\n"
-> "X-Pact-Broker-Git-Sha: ed634ba7507dc130125d95ca8d5acc2b7bd28f23\r\n"
-> "X-Pact-Broker-Version: 2.23.1\r\n"
-> "X-Powered-By: Phusion Passenger 5.0.29\r\n"
-> "Content-Length: 939\r\n"
-> "Connection: Close\r\n"
-> "\r\n"
reading 939 bytes...
-> "{\"summary\":{\"deployable\":true,\"reason\":\"All verification results are published and successful\"},\"matrix\":[{\"consumer\":{\"name\":\"de.tine.foo\",\"version\":{\"number\":\"3.0.0\",\"_links\":{\"self\":{\"href\":\"https://test.pact.dius.com.au/pacticipants/de.tine.foo/versions/3.0.0\"}}},\"_links\":{\"self\":{\"href\":\"https://test.pact.dius.com.au/pacticipants/de.tine.foo\"}}},\"provider\":{\"name\":\"de.tine.baz\",\"version\":{\"number\":\"20.0.0\"},\"_links\":{\"self\":{\"href\":\"https://test.pact.dius.com.au/pacticipants/de.tine.baz\"}}},\"pact\":{\"createdAt\":\"2018-07-01T21:24:22+10:00\",\"_links\":{\"self\":{\"href\":\"https://test.pact.dius.com.au/pacts/provider/de.tine.baz/consumer/de.tine.foo/version/3.0.0\"}}},\"verificationResult\":{\"success\":true,\"verifiedAt\":\"2018-07-01T21:23:28+10:00\",\"_links\":{\"self\":{\"href\":\"https://test.pact.dius.com.au/pacts/provider/de.tine.baz/consumer/de.tine.foo/pact-version/b74c6ff8b6c7b91375dc0f6a17f1066e6fd72250/verification-results/4952\"}}}}]}"
read 939 bytes
Conn close
Computer says yes \o/ 

CONSUMER    | C.VERSION | PROVIDER    | P.VERSION | SUCCESS?
------------|-----------|-------------|-----------|---------
de.tine.foo | 3.0.0     | de.tine.baz | 20.0.0    | true    

All verification results are published and successful
```

The `GET /matrix` request doesn't contain an entry for `de.tine.bar`:

Request:
````
curl  'https://test.pact.dius.com.au/matrix?q%5B%5D%5Bpacticipant%5D=de.tine.foo&q%5B%5D%5Bversion%5D=3.0.0&latestby=cvp&latest=true&tag=prod'  -u <snip>
````

Response:
````
{
	"summary": {
		"deployable": true,
		"reason": "All verification results are published and successful"
	},
	"matrix": [{
		"consumer": {
			"name": "de.tine.foo",
			"version": {
				"number": "3.0.0",
				"_links": {
					"self": {
						"href": "https://test.pact.dius.com.au/pacticipants/de.tine.foo/versions/3.0.0"
					}
				}
			},
			"_links": {
				"self": {
					"href": "https://test.pact.dius.com.au/pacticipants/de.tine.foo"
				}
			}
		},
		"provider": {
			"name": "de.tine.baz",
			"version": {
				"number": "20.0.0"
			},
			"_links": {
				"self": {
					"href": "https://test.pact.dius.com.au/pacticipants/de.tine.baz"
				}
			}
		},
		"pact": {
			"createdAt": "2018-07-01T21:24:22+10:00",
			"_links": {
				"self": {
					"href": "https://test.pact.dius.com.au/pacts/provider/de.tine.baz/consumer/de.tine.foo/version/3.0.0"
				}
			}
		},
		"verificationResult": {
			"success": true,
			"verifiedAt": "2018-07-01T21:23:28+10:00",
			"_links": {
				"self": {
					"href": "https://test.pact.dius.com.au/pacts/provider/de.tine.baz/consumer/de.tine.foo/pact-version/b74c6ff8b6c7b91375dc0f6a17f1066e6fd72250/verification-results/4952"
				}
			}
		}
	}]
}
````

**Comment by bethesque** [comment]
Thanks for working out how to recreate the issue. I'll have a look at what the code is doing.

**Comment by bethesque** [comment]
That was an interesting one! I've fixed it and released v2.23.4 (docker 2.23.4-1)

**Comment by tinexw** [comment]
Thanks for the quick fix, I upgraded and it works as expected now :-)
